chihiro_todo/chihiro_jquery_todo/jquery.js

```diff
@@ -0,0 +1,190 @@
+        taskPersonElm.before(todoButton);
+        todoButton.click(changeToTodo); 
+    }
+
+    if (taskArea[0].id !== "doing_area") {
+        const doingButton =$('<button>').text("doing");
+        taskPersonElm.before(doingButton);
+        doingButton.click(changeToDoing);
+    }
+
+    if (taskArea[0].id !== "done_area") {
+        const doneButton =$('<button>').text("done");
+        taskPersonElm.before(doneButton);
+        doneButton.click(changeToDone);
+    }
+
+    const deleteButton =$('<button>').text("delete");
+    taskPersonElm.before(deleteButton)
+    deleteButton.click(deleteTask);
+}
+
+//タスクのステータスをtodoに変更する関数
+function changeToTodo() {
+    const targetTaskElm = $(this).closest('li');
+    targetTaskElm.appendTo($('#todo_area'))
+    appendButtonByStatus(targetTaskElm);
+}
+
+//タスクのステータスをdoingに変更する関数
+function changeToDoing() {
+    const targetTaskElm = $(this).closest('li');
+    targetTaskElm.appendTo($('#doing_area'));
+    appendButtonByStatus(targetTaskElm);
+}
+
+//タスクのステータスをdoneに変更する関数
+function changeToDone() {
+    const targetTaskElm =$(this).closest('li');
+    targetTaskElm.appendTo($('#done_area'));
+    appendButtonByStatus(targetTaskElm);
+}
+
+//タスクを削除する関数
+function deleteTask() {
+    const targetTaskElm = $(this).closest('li');
+    targetTaskElm.remove();
+}
+
+//担当者に削除ボタンを付与する関数
+function deletePersonButton(personElm) {
+    const deleteButton =$('<button>').text("delete");
+    deleteButton.appendTo(personElm);
+    deleteButton.click(deletePerson);
+}
+
+//担当者をPersonListとselectboxから削除する関数
+function deletePerson() {
+    const personLiElm = $(this).closest('li');
+    const personLiId =personLiElm.data('person-id').toString()
+    personLiElm.remove();
+    $('#person_select option').each(function() {
+        const selectOptionId =$(this).val();
+        const targetPersonElm = $(this).closest('option')
+        if (selectOptionId === personLiId) {
+           targetPersonElm.remove()
+        }
+    })
+}
+
+//担当者に編集ボタンを付与する関数
+function editPersonButton(personElm) {
+    const editButton =$('<button>').text("edit");
+    editButton.appendTo(personElm);
+    editButton.click(function() {
+        appendEditInput(personElm,editButton);
+    })
+}
+
+function appendEditInput(personElm, editButton) {
+    const beforeChangePersonElm = personElm.children("span")
+    const beforeEditPersonName = beforeChangePersonElm.text()
+    if (underEditedPersonId !== undefined) {
+        $('#person_list_area li').each(function() {
+            if (($(this).data('person-id') === underEditedPersonId)) {
+            const cancelEditPersonElm = $(this)
+            cancelEditPerson(cancelEditPersonElm.children("span").text(), cancelEditPersonElm)
+            }
+        })
+    }
+    underEditedPersonId = personElm.data('person-id');
+    const editPersonInput =$('<input>').attr('id', 'edit_input')
+    editPersonInput.val(beforeEditPersonName);
+    editPersonInput.replaceAll(beforeChangePersonElm)
```

Comment:
@chihiro-kakizaki 
全体的に、こんな感じでメソッドチェーンで書いてあげると、変数名考える地獄から解放されるかも！！笑
```suggestion
     $('<input>').attr('id', 'edit_input')
        .val(beforeEditPersonName)
        .replaceAll(beforeChangePersonElm);
```

Reply:
@Shibata0806 
メソッドチェーンを使うことで変数名の定義が要らなくなるのですね！
以下のコミットで修正しました。
[メソッドチェーンに修正](https://github.com/chihiro-kakizaki/todolist/pull/5/commits/dc5fb466c6bfbae2a295bcea9120afeb28796a64)

各ボタンについてはメソッドチェーンを使うか迷いましたが、コードの解読が難しくなるかなと考え、メソッドチェーンを使わずにそのまま記述しています。
気になる点がありましたらご指摘頂けると幸いです🙇‍♀️